pkg/controllers/disruption/helpers.go

```diff
@@ -191,6 +191,45 @@ func GetCandidates(ctx context.Context, cluster *state.Cluster, kubeClient clien
 	return lo.Filter(candidates, func(c *Candidate, _ int) bool { return shouldDeprovision(ctx, c) }), nil
 }
 
+// BuildDisruptionBudgets will return a map for nodePoolName -> numAllowedDisruptions and an error
+func BuildDisruptionBudgets(ctx context.Context, cluster *state.Cluster, clk clock.Clock, kubeClient client.Client) (map[string]int, error) {
+	nodePoolList, err := nodepoolutil.List(ctx, kubeClient)
+	if err != nil {
+		return nil, fmt.Errorf("listing node pools, %w", err)
+	}
+	numNodes := map[string]int{}
+	deleting := map[string]int{}
+	disruptionBudgetMapping := map[string]int{}
+	// We need to get all the nodes in the cluster
+	// Get each current active number of nodes per nodePool
+	// Get the max disruptions for each nodePool
+	// Get the number of deleting nodes for each of those nodePools
+	// Find the difference to know how much left we can disrupt
+	nodes := cluster.Nodes()
+	for _, node := range nodes {
+		if !node.Managed() {
+			continue
+		}
+		nodePool := node.Labels()[v1beta1.NodePoolLabelKey]
+		if node.MarkedForDeletion() {
+			deleting[nodePool]++
+		}
+		numNodes[nodePool]++
+	}
+
+	for i := range nodePoolList.Items {
+		nodePool := nodePoolList.Items[i]
+		disruptions, err := nodePool.GetAllowedDisruptions(ctx, clk, numNodes[nodePool.Name])
+		if err != nil {
+			// Don't return here as we shouldn't block all disruption within the cluster on one misconfigured nodepool.
+			logging.FromContext(ctx).With("nodePool", nodePool.Name).Errorf("getting allowed disruptions, %w", err)
+		}
+		// Subtract the allowed number of disruptions from the number of already deleting nodes.
+		disruptionBudgetMapping[nodePool.Name] = lo.Clamp(disruptions-deleting[nodePool.Name], 0, math.MaxInt32)
```

Comment:
Sidenote: I think we know about this implicitly because this is how PDB works but we may want to clarify how we count "disruptions" per cycle. For example, this code would count a node in deleting state that was marked for deletion in the prior disruption cycle and will continue to drain the budget until it resolves. I believe this is the same behavior as PDB but I don't think it was document explicitly.

Reply:
We'll definitely document this explicitly in the website. Would you like to see this comment expanded? 

I was hoping to drive the definition of this through the API comment itself here: https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1beta1/nodepool.go#L104-L112 but I'm realizing this comment is not too descriptive and partly wrong, so I'll make sure to update this too.

Reply:
```
// Nodes dictates how many NodeClaims owned by this NodePool
// can be terminating at once
``` 
	
I think is just a little to broad/ up for interpretation because what we are actually doing isn't terminating all these node at once and some nodes can take longer to terminate. I think should be something along the lines of `maximum number of nodes a node pool can tolerate being disrupted`

Reply:
Will adjust. Thanks for the feedback!